test/internal/util/transcation_id_generator_test.dart

```diff
@@ -0,0 +1,9 @@
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter_ble_lib/internal/util/transaction_id_generator.dart';
+
+void main() {
+  test("should be able to generate an id", () {
+    expect(TransactionIdGenerator.getNextId(), isNotNull);
+  });
+}
```

Comment:
Maybe also check if next call will not return the previous result?

Reply:
I added a test that checks if next invocation returns incremented id so it also covers your case.

Reply:
The generator needs more test but this time I added test only for the fixed scenario with full determination.